src/app/static/src/app/store/plotSelections/utils.ts

```diff
@@ -0,0 +1,145 @@
+import { Commit } from "vuex";
+import { CalibrateMetadataResponse, FilterOption, FilterRef, FilterTypes, PlotSettingOption } from "../../generated";
+import { PlotName } from "./plotSelections";
+import { PlotSelectionUpdate, PlotSelectionsMutations } from "./mutations";
+import { RootState } from "../../root";
+
+export const filtersAfterUseShapeRegions = (filterTypes: FilterTypes[], rootState: RootState) => {
+    const filters = [...filterTypes];
+    const area = filters.find(f => f.id == "area");
+    if (area && area.use_shape_regions) {
```

Comment:
I make a couple of changes to this in my branch but I think no need to check filter type here is "area". If the flag is set for `use_shape_regions` then I think we should switch it out.